### DISCUSSION TITLE
v8 alpha.24: How to derive the Row type?

### DISCUSSION THREAD

**Discussion by Obiwarn** [TanStack/table#3818] [Q&A] [answered]
Hey there,

I still have a hard time to learn all the new typing in v8.

Let's say I want to add an `onClick` handler to the rows in the basic example:

```
  const table = createTable<{ Row: Person }>();

  const handleClickRow = (row: Row< XXX >) => {
     // ...
  };
```

How do I get XXX? 

Substituting XXX with `{ Row: Person }` gives me a type error.

 I guess I can derive it from `table`?

Does anyone have an idea?

**Comment by tannerlinsley** [comment, marked as answer]
This should get you on the right track: https://codesandbox.io/s/flamboyant-resonance-k503o8?file=/src/main.tsx

**Reply by Obiwarn** [reply]
Indeed. Thank you so much.

For others:
The gist to get the generics:

```
const table = createTable<{ Row: Person }>();

type TableGenerics = typeof table.generics;
```
